lib/package/dynamodb/compileMethodsToDynamodb.js

```diff
@@ -0,0 +1,387 @@
+        case 'POST':
+          return 'PutItem'
+        case 'GET':
+          return 'GetItem'
+        case 'DELETE':
+          return 'DeleteItem'
+        case 'PATCH':
+          return 'UpdateItem'
+      }
+    }
+
+    return http.action
+  },
+
+  getDynamodbMethodIntegration(http) {
+    const integration = {
+      IntegrationHttpMethod: 'POST',
+      Type: 'AWS',
+      Credentials: {
+        'Fn::GetAtt': ['ApigatewayToDynamodbRole', 'Arn']
+      },
+      Uri: {
+        'Fn::Sub': [
+          'arn:aws:apigateway:${AWS::Region}:dynamodb:action/${action}',
+          { action: http.action }
+        ]
+      },
+      PassthroughBehavior: 'NEVER',
+      RequestTemplates: this.getDynamodbIntegrationRequestTemplates(http)
+    }
+
+    const integrationResponse = {
+      IntegrationResponses: [
+        {
+          StatusCode: 200,
+          SelectionPattern: '2\\d{2}',
+          ResponseParameters: {},
+          ResponseTemplates: this.getDefaultDynamodbResponseTemplates(http)
+        },
+        {
+          StatusCode: 400,
+          SelectionPattern: '4\\d{2}',
+          ResponseParameters: {},
+          ResponseTemplates: {}
+        },
+        {
+          StatusCode: 500,
+          SelectionPattern: '5\\d{2}',
+          ResponseParameters: {},
+          ResponseTemplates: {}
+        }
+      ]
+    }
+
+    if (http && http.cors) {
+      let origin = http.cors.origin
+      if (http.cors.origins && http.cors.origins.length) {
+        origin = http.cors.origins.join(',')
+      }
+
+      integrationResponse.IntegrationResponses.forEach(async (val, i) => {
+        integrationResponse.IntegrationResponses[i].ResponseParameters = {
+          'method.response.header.Access-Control-Allow-Origin': `'${origin}'`
+        }
+      })
+    }
+
+    _.merge(integration, integrationResponse)
+
+    return {
+      Properties: {
+        Integration: integration
+      }
+    }
+  },
+
+  getDynamodbIntegrationRequestTemplates(http) {
+    const defaultRequestTemplates = this.buildDefaultDynamodbRequestTemplates(http)
+    return Object.assign(defaultRequestTemplates, _.get(http, ['request', 'template']))
+  },
+
+  buildDefaultDynamodbRequestTemplates(http) {
+    return {
+      'application/json': this.buildDefaultDynamodbRequestTemplate(http),
+      'application/x-www-form-urlencoded': this.buildDefaultDynamodbRequestTemplate(http)
+    }
+  },
+
+  getDynamodbObjectHashkeyParameter(http) {
+    if (http.hashKey.pathParam) {
+      return {
+        key: http.hashKey.pathParam,
+        attributeType: http.hashKey.attributeType,
+        attributeValue: `$input.params().path.${http.hashKey.pathParam}`
+      }
+    }
+
+    if (http.hashKey.queryStringParam) {
+      return {
+        key: http.hashKey.queryStringParam,
+        attributeType: http.hashKey.attributeType,
+        attributeValue: `$input.params().querystring.${http.hashKey.queryStringParam}`
+      }
+    }
+
+    if (http.hashKey.bodyParam) {
```

Comment:
It this allowed in the schema?
Aren't body params used to pass in update expression, etc.?